packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -44,6 +44,11 @@ abstract class OperatingSystemUtils {
   File makePipe(String path);
 
   void unzip(File file, Directory targetDirectory);
+
+  /// Returns the name of the [binaryName] executable.
+  /// 
+  /// No-op on most OS. Adds [winExtension] to the name on Windows.
```

Comment:
Make winExtension default to `exe` in the method signature (here and in the implementations below) so that it's clear the extension will be added for Windows even if the caller doesn't specify it.